core/src/dird/ndmp_dma_storage.cc

```diff
@@ -997,19 +997,17 @@ bool NdmpSendLabelRequest(UaContext* ua,
 static bool ndmp_native_update_runtime_storage_status(JobControlRecord* jcr,
                                                       StorageResource* store)
 {
-  bool retval = true;
-
-  lock_mutex(store->runtime_storage_status->changer_lock);
-  if (!do_ndmp_native_query_tape_and_robot_agents(jcr, store)) {
-    retval = false;
+  {
+    const std::lock_guard lock_(store->runtime_storage_status->changer_lock);
+    if (!do_ndmp_native_query_tape_and_robot_agents(jcr, store)) {
+      return false;
+    }
   }
-  unlock_mutex(store->runtime_storage_status->changer_lock);
 
-  lock_mutex(store->runtime_storage_status->changer_lock);
-  if (!NdmpUpdateStorageMappings(jcr, store)) { retval = false; }
-  unlock_mutex(store->runtime_storage_status->changer_lock);
+  const std::lock_guard lock_(store->runtime_storage_status->changer_lock);
+  if (!NdmpUpdateStorageMappings(jcr, store)) { return false; }
 
-  return retval;
+  return true;
```

Comment:
Hmm.... are we sure that we want to change the behaviour here ?  Previously `NdmpUpdateStorageMappings` was called regardless of whether the first function succeeded or not.

Reply:
Definitely not. Good catch, I changed it back.